### PR TITLE
backupccl: add checks for a dropped full schedule

### PR DESCRIPTION
**Pull request by benbardin**
Release justification: High-impact bugfix (NPE).

Release note (enterprise change): Fixes NPE when ALTER BACKUP SCHEDULE
was called after a dependent schedule was dropped.

### CODE REVIEW

File: pkg/ccl/backupccl/schedule_exec.go [hunk 1]
```diff
@@ -439,7 +439,9 @@ func unlinkDependentSchedule(
 
 	// Clear the DependentID field since we are dropping the record associated
 	// with it.
+	// TODO(benbardin): This is probably surprising behavior, rethink.
```
Comment (adityamaru) [inline pkg/ccl/backupccl/schedule_exec.go:442]: Should we reword this TODO with a link to https://github.com/cockroachdb/cockroach/issues/87435 instead? As is this TODO isn't very informative so we should flesh it out depending on what you decide.

File: pkg/ccl/backupccl/alter_backup_schedule.go [hunk 2]
```diff
@@ -115,6 +115,12 @@ func doAlterBackupSchedules(
 		return err
 	}
 
+	if s.fullJob == nil {
+		// This can happen if a user calls DROP SCHEDULE on the full schedule.
+		// TODO(benbardin): Can we prevent this state from occurring in the first place?
```
Comment (adityamaru) [inline pkg/ccl/backupccl/alter_backup_schedule.go:120]: Can we link https://github.com/cockroachdb/cockroach/issues/87435 as the desired state in this TODO. I think this would prevent this state from showing up.

File: pkg/ccl/backupccl/alter_backup_schedule.go [hunk 3]
```diff
@@ -115,6 +115,12 @@ func doAlterBackupSchedules(
...
+		return errors.Newf("no full backup schedule found, drop this schedule and recreate")
```
Comment (adityamaru) [inline pkg/ccl/backupccl/alter_backup_schedule.go:121]: IIUC this case can only occur if a full is dropped, and the user alters its inc?

In that case maybe this would be a clearer call to action to the user:

"schedule %d does not have a corresponding full backup schedule; drop schedule %d and recreate"

File: pkg/ccl/backupccl/testdata/backup-restore/alter-schedule/missing-schedule [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+with schedules as (show schedules) select id, state, recurrence, owner, command from schedules where label='datatest' order by command->>'backup_type' asc;
+----
+$fullID <nil> @weekly root {"backup_statement": "BACKUP INTO 'nodelocal://1/example-schedule' WITH detached", "chain_protected_timestamp_records": true, "dependent_schedule_id": $incID, "unpause_on_success": $incID}
+$incID Waiting for initial backup to complete @daily root {"backup_statement": "BACKUP INTO LATEST IN 'nodelocal://1/example-schedule' WITH detached", "backup_type": 1, "chain_protected_timestamp_records": true, "dependent_schedule_id": $fullID}
+
+exec-sql
+drop schedule $incID
+----
+
+# The schedule is now in an invalid state.
```
Comment (adityamaru) [inline pkg/ccl/backupccl/testdata/backup-restore/alter-schedule/missing-schedule:60]: nit: this is not an invalid state? inc being dropped is an acceptable state to be in.

**Review by adityamaru — APPROVED** [review]
LGTM, only nits

File: pkg/ccl/backupccl/alter_backup_schedule.go (hunk 2 above)
Comment (benbardin) [reply pkg/ccl/backupccl/alter_backup_schedule.go]: Yeah, that's a good idea!

File: pkg/ccl/backupccl/alter_backup_schedule.go (hunk 3 above)
Comment (benbardin) [reply pkg/ccl/backupccl/alter_backup_schedule.go]: I like this, thanks!

File: pkg/ccl/backupccl/schedule_exec.go (hunk 1 above)
Comment (benbardin) [reply pkg/ccl/backupccl/schedule_exec.go]: Yup, fixed.

File: pkg/ccl/backupccl/testdata/backup-restore/alter-schedule/missing-schedule (hunk 4 above)
Comment (benbardin) [reply pkg/ccl/backupccl/testdata/backup-restore/alter-schedule/missing-schedule]: Copy-pasta, fixed. Thansk!
